commit 6836d0674efdb1a6b79953285f10d8edd7e20432
Author: Jeremy Evans <code@jeremyevans.net>
Date:   Tue Mar 5 21:57:34 2019 -0800

    unicorn_rails: fix regression with Rails >= 3.x in app build
    
    Note: `unicorn_rails' was only intended for Rails <= 2.x projects
    in the old days.
    
    Fixes: 5985dd50a9bd7238 ("Support default_middleware configuration option")
    
    From: Jeremy Evans <code@jeremyevans.net>
      cf. https://bogomips.org/unicorn-public/20190306055734.GC61406@jeremyevans.local/
    Signed-off-by: Eric Wong <e@80x24.org>
      [ew: commit message]

diff --git a/bin/unicorn_rails b/bin/unicorn_rails
index ea4f822..354c1df 100755
--- a/bin/unicorn_rails
+++ b/bin/unicorn_rails
@@ -132,11 +132,11 @@ def rails_builder(ru, op, daemonize)
 
   # this lambda won't run until after forking if preload_app is false
   # this runs after config file reloading
-  lambda do ||
+  lambda do |x, server|
     # Rails 3 includes a config.ru, use it if we find it after
     # working_directory is bound.
     ::File.exist?('config.ru') and
-      return Unicorn.builder('config.ru', op).call
+      return Unicorn.builder('config.ru', op).call(x, server)
 
     # Load Rails and (possibly) the private version of Rack it bundles.
     begin
